### ISSUE TITLE
Unable to use idp_hint with google idp

### ISSUE THREAD

**Issue by mooreds** [FusionAuth/fusionauth-issues#909]
### Description 

I configured google as an identity provider. I can login with google if I go directly to the login page and click the link. If, however, I add an `idp_hint` as documented [here](https://fusionauth.io/docs/v1/tech/release-notes#version-1-19-0), I get an error message

```
We're sorry, your request was malformed or was unable to be completed for some reason. Try hitting the back button and restarting the process to see if it fixes the problem. 
```

Here's the URL I visit: https://fusionauthurl/oauth2/authorize?client_id=5a6a3ee2-429b-4f66-bc70-1f9a96892eb2&response_type=code&redirect_uri=https%3A%2F%2Fpiedpiper.com&idp_hint=82339786-3dff-42a6-aac6-1f1ceecb6c46 

### Affects versions
1.19.7

### Steps to reproduce
1. Configure the google idp as documented
1. Note the google idp uuid
1. Enable it for an application.
1. View the application and note the login url 
1. Append idp_hint=uuid
1. Visit the url in an incognito window.
1. See error message above.

### Expected behavior

You should be directed to the google hosted signin form.

### Internal Notes

I think we need to make Google and Facebook configurable as a popup or a redirect. This requires some JavaScript updates, some testing, updating documentation for configuration, etc. 

If we add the option to use a redirect for a more traditional OAuth/OIDC login for Google and FB then the `idp_hint` should work. 

### Doc
- Update notes on hints once complete. https://fusionauth.io/docs/v1/tech/identity-providers/

### Additional context

Reported in this forum post: https://fusionauth.io/community/forum/topic/447/malformed-request-when-using-idp_hint

**Comment by mooreds** [comment]
Is `idp_hint` limited to non social providers?

**Comment by mooreds** [comment]
Also added some documentation about this feature: https://github.com/FusionAuth/fusionauth-site/pull/231

**Comment by robotdan** [comment]
I added some comments to  FusionAuth/fusionauth-site#231 - not sure this will work with Google, it uses a modal dialog. We'd have to see if the Google API allows us to hint, and then ensure it gets into the right spot.

**Comment by mooreds** [comment]
Actually, we kind of have this already with the complete login api calls: https://fusionauth.io/docs/v1/tech/apis/identity-providers/facebook#complete-the-facebook-login

**Comment by trevorr** [comment]
Just when I thought `idp_hint` solved all my problems, I stumbled across this issue. 😅 It doesn't seem to work for Apple or Facebook either. I'd like to use my own login UI with social buttons, since I'm using Material UI and can't easily replicate its look and feel as a FusionAuth theme.

I see that this issue is In progress. Is a fix likely to be released soon, or is there another workaround? The "complete the login" APIs seem like they require implementing all the third-party API interactions myself, which is definitely something I'd rather let FA keep doing.

**Comment by trevorr** [comment]
> If we add the option to use a redirect for a more traditional OAuth/OIDC login for Google and FB then the `idp_hint` should work.

Also note that redirects seem to be more reliable in the presence of browser extensions like [Privacy Badger](https://privacybadger.org/). If I have that extension enabled, FA Google login doesn't work.

**Comment by robotdan** [comment]
We have this mostly coded and should be out soon. The initial fix will be to allow Google and Facebook to be configured to use a redirect instead of a pop up. This then should allow the idp_hint to work.

I was thinking Apple would work already, but perhaps not. We can take a look at that one as well as part of this effort.

**Comment by trevorr** [comment]
Awesome, thanks! For whatever reason, even though it doesn't use pop-ups, I get the same malformed request error for Apple. All three work fine from the FA login UI.

**Comment by robotdan** [comment]
@trevorr  It looks like the Apple one is using a JS popup - but it is not obvious to the end user. When in Safari it does a native modal login, and on other browsers it does perform a redirect. But it is currently all initiated from the Apple `AppleID.auth.init` JS. 

So we may have some work to do in order to support the `idp_hint` with the Apple provider. 

@brettwp we can likely ship the work we have shortly, but once we get some free cycles we should take a look at the Apple IdP as well to see what is necessary to optionally use a regular redirect to support the `idp_hint`. Not even sure if Apple will support this but we should take a look.

**Comment by robotdan** [comment]
Ok, great, that is good to know. 

For your awareness, once this ships - the way this will work is you'll have an additional toggle in the Google / FB IdP config to ask for a popup or redirect style login flow. 

This is primarily to allow us to be backwards compatible, if we had to do it over, we'd probably just offer the redirect and skip the popup. But.. hindsight is 20/20. :-) 

So once you config the redirect vs popup (and add additional config to Google and FB to register the redirect) - then then `idp_hint` should "just work" ™.

**Comment by robotdan** [comment]
This is available in 1.28.0. We need to still look at Apple and Twitter, and maybe others (?) to see if any other IdPs need to be re-worked to support `idp_hint` with a redirect.

**Comment by robotdan** [comment]
Closing this one out. Opened https://github.com/FusionAuth/fusionauth-issues/issues/1306 to track Apple and any other IdPs we do not yet support for `idp_hint`.
